src/WebJobs.Script/RoutingUtility.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Constraints;
+using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.AspNetCore.Routing.Tree;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public static class RoutingUtility
+    {
+        private readonly static IDictionary<string, TemplateMatcher> Templates = new Dictionary<string, TemplateMatcher>();
+        private readonly static IDictionary<string, IDictionary<string, IRouteConstraint>> ConstraintsMap = new Dictionary<string, IDictionary<string, IRouteConstraint>>();  
+        private readonly static IOptions<RouteOptions> RouteOptions = new OptionsWrapper<RouteOptions>(new RouteOptions());
+        private readonly static IRouter Router = new FakeRouter();
+        private readonly static IInlineConstraintResolver ConstraintResolver = new DefaultInlineConstraintResolver(RouteOptions);
+        private readonly static AspNetCore.Http.HttpContext Context = new DefaultHttpContext();
+
+        //a dictionary of route constraints to functions that preform the appropriate conversions
+        private static readonly IDictionary<Type, Func<string, object>> StringConverter = new Dictionary
+            <Type, Func<string, object>>
+        {
+            {typeof(BoolRouteConstraint), (str => Convert.ToBoolean(str)) },
+            {typeof(DateTimeRouteConstraint), (str => Convert.ToDateTime(str)) },
+            {typeof(DecimalRouteConstraint), (str => Convert.ToDecimal(str)) },
+            {typeof(DoubleRouteConstraint), (str => Convert.ToDouble(str)) },
+            {typeof(FloatRouteConstraint), (str => Convert.ToSingle(str)) },
+            {typeof(GuidRouteConstraint), (str => Guid.Parse(str)) },
+            {typeof(IntRouteConstraint), (str => Convert.ToInt32(str)) },
+            {typeof(LongRouteConstraint), (str => Convert.ToInt64(str)) },
+            {typeof(RangeRouteConstraint), (str => Convert.ToInt64(str)) }
+        };  
+
+        private static TemplateMatcher GetTemplateMatcher(string queryTemplate)
```

Comment:
This method seems to be doing 2 things, 1)returning a TemplateMatcher, as well as adding to the Templates list.  It has an unexpected hidden side-effect.  Should you bring out 

Templates.Add(queryTemplate, templateMatcher);

where templateMatcher is added to the Template "after" this method is called?